### PR TITLE
Add test case for syslog rate limit feature

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
HLD: https://github.com/sonic-net/SONiC/pull/1049

Change-Id: I11f29fa1a2265e2ba8c9a4ea4bac6ed2a0bb5a25

### Description of PR
Add test case for syslog rate limit feature.

Summary:
test_syslog_rate_limit_container

1.	Randomly choose on container
2.	Change the syslog rate limit of current container
3.	Use a generated script to print log from current container which is fast enough to hit the limit
4.	Check syslog that some logs are dropped

test_syslog_rate_limit_host

1.	Change the syslog rate limit of host
2.	Use a generated script to print log from host which is fast enough to hit the limit
3.	Check syslog that some logs are dropped

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?

Add test case for syslog rate limit feature

#### How did you do it?

Add test case for syslog rate limit feature

#### How did you verify/test it?

Run the test case

#### Any platform specific information?

N/A

#### Supported testbed topology if it's a new test case?

N/A

### CODE REVIEW

**Review by keboliu — APPROVED** [review]

File: tests/syslog/test_syslog_rate_limit.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+def check_image_version(rand_selected_dut):
+    """Skips this test if the SONiC image installed on DUT is older than 202205
+
+    Args:
+        duthost: Hostname of DUT.
+
+    Returns:
+        None.
+    """
+    skip_release(rand_selected_dut, ["201811", "201911", "202012", "202106", "202205"])
```
Comment (wangxin) [inline tests/syslog/test_syslog_rate_limit.py:42]: Can you use the conditional mark plugin to do the skipping?

Comment (Junchao-Mellanox) [reply tests/syslog/test_syslog_rate_limit.py:42]: I did not find such conditional mark plugin. Do you mean to create one?

File: tests/syslog/test_syslog_rate_limit.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    # Copy tests/syslog/log_generator.py to DUT
+    rand_selected_dut.copy(src=LOCAL_LOG_GENERATOR_FILE, dest=REMOTE_LOG_GENERATOR_FILE)
+
+    verify_container_rate_limit(rand_selected_dut)
+    verify_host_rate_limit(rand_selected_dut)
+
+    # Save configuration and reload, verify the configuration can be loaded
+    logger.info('Persist syslog rate limit configuration to DB and do config reload')
+    rand_selected_dut.command('config save -y')
+    config_reload(rand_selected_dut)
```
[the remark this answers is not in the document]
Comment (Junchao-Mellanox) [reply tests/syslog/test_syslog_rate_limit.py:81]: I did not find any complain due my test.

**Comment by Junchao-Mellanox** [comment]
Hi @wangxin, i have replied to your comment, would you please check if it addressed your comment?

**Comment by liat-grozovik** [comment]
@Junchao-Mellanox could we move forward merging this one now that the feature PRs are all in? or should we wait for some submodule updates?

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**